### ISSUE TITLE
Folders in explorer disappear when clicked

### ISSUE THREAD

**Issue by SRv6d** [microsoft/vscode#231304]
Type: <b>Bug</b>

1. Clone FRRouting master branch: https://github.com/FRRouting/frr (d1433ee9a8fa374ee653dcf1fe6e852481b17119)

2. Try to open directories like bgpd, bfdd etc.

VS Code version: Code 1.94.2 (384ff7382de624fb94dbaf6da11977bba1ecd427, 2024-10-09T16:08:44.566Z)
OS version: Darwin arm64 24.0.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Max (10 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 5, 5|
|Memory (System)|32.00GB (1.17GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (34)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-preview-github-styles|bie|2.1.0
ruff|cha|2024.50.0
gitlens|eam|15.6.0
EditorConfig|Edi|0.16.4
prettier-vscode|esb|11.0.0
dependi|fil|0.7.10
remotehub|Git|0.64.0
vscode-github-actions|git|0.27.0
vscode-pull-request-github|Git|0.98.0
go|gol|0.42.1
fluent-icons|mig|0.0.18
symbols|mig|0.0.20
vscode-docker|ms-|1.29.3
debugpy|ms-|2024.10.0
python|ms-|2024.16.1
vscode-pylance|ms-|2024.10.1
jupyter|ms-|2024.9.1
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.19
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-containers|ms-|0.388.0
remote-ssh|ms-|0.115.0
remote-ssh-edit|ms-|0.87.0
remote-wsl|ms-|0.88.4
vscode-remote-extensionpack|ms-|0.25.0
azure-repos|ms-|0.40.0
live-server|ms-|0.4.15
remote-explorer|ms-|0.4.3
remote-repositories|ms-|0.42.0
remote-server|ms-|1.5.2
rust-analyzer|rus|0.3.2146
jinjahtml|sam|0.20.0
even-better-toml|tam|0.19.2

(1 theme extensions excluded)

</details>

**Comment by lramos15** [comment]
![Image](https://github.com/user-attachments/assets/2ee01970-ca80-4f97-b851-4d9bf8b2a74a)
I recommend doing `F1 -> Troubleshoot Issue`

**Comment by SRv6d** [comment]
Issue troubleshooting told me it was my configuration, from there I narrowed it down to the `"explorer.excludeGitIgnore": true,` setting. If I set that to false the issue goes away.

**Comment by lramos15** [comment]
Interesting. And this is just off main, so you haven't edited the .gitignore at all?

**Comment by SRv6d** [comment]
This is just on main, without touching anything. The files are not greyed out when the setting is set to false.
Interestingly, the problem persists with the setting on even when I remove the `.gitignore`.

**Comment by SRv6d** [comment]
So the nested `.gitignores` are the culprit. https://github.com/FRRouting/frr/blob/40dce0be6a8f1cad15fce434177b48f9e69b96dd/bfdd/.gitignore for example, inside of the bfdd directory, containing `bfdd`.
I'm guessing vscode loads the nested .gitignore once the directory is opened and applies it globally instead of recursively just inside of that directory. I'm not sure about the intentions of this kind of layout but if I'm not mistaken they should be applied recursively to the directory they were found in.

**Comment by SRv6d** [comment]
@sourcefrog Thanks for testing and adding the relevant git documentation!

**Comment by lramos15** [comment]
Yes this is a bug on our end and I have marked it as such. Thank you for reporting it.

Due to not all machines having the git executable and to support places like web we reimplement the gitignore functionality
